src/tools/rbd_mirror/MirrorStatusUpdater.cc

```diff
@@ -302,12 +300,6 @@ void MirrorStatusUpdater<I>::update_task(int r) {
     librados::ObjectWriteOperation op;
     uint32_t op_count = 0;
 
-    if (!m_site_name.empty()) {
-      // updates to remote sites should include local site name
-      // to ensure status includes this peer
-      librbd::cls_client::mirror_peer_ping(&op, m_site_name, m_fsid);
-    }
-
```

Comment:
Just wondering, isn't a situation possible, when `update_task` is fired from the timer with empty `m_updating_global_image_ids`, resulting in  `aio_operate` trigger assert due to empty `op`?

It seems if it had been possible we would have already observed this for local mirror status updater, which we have not. On the other hand I can't see what prevents this theoretical possibility on an idle rbd-mirror.

Reply:
Thanks. I appended on a commit to test for this and skip the rados op.